### ISSUE TITLE
train test alpha

### ISSUE THREAD

**Issue by tgroechel** [interaction-lab/BehaviorTreeFromDemonstration#39]

**Comment by tgroechel** [comment]
Need to:
- [ ] add proper k-fold for average train/test accuracy for pruning
- [ ] properly label training error for finalized models
- [ ] maybe add validation set for hyperparam optimization but likely to make the a whole new issue

**Comment by tgroechel** [comment]
This looks promising: https://medium.com/@sanchitamangale12/decision-tree-pruning-cost-complexity-method-194666a5dd2f

**Comment by tgroechel** [comment]
Closing with new k-fold implemented. moving validation tuning to another issue
